clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1808,34 +1808,6 @@ void Clang::AddAArch64TargetArgs(const ArgList &Args,
 
   AddUnalignedAccessWarning(CmdArgs);
 
-  Args.addOptInFlag(CmdArgs, options::OPT_fptrauth_intrinsics,
```

Comment:
https://maskray.me/blog/2023-08-25-clang-wunused-command-line-argument#target-specific-options

You can make these options TargetSpecific and only handle them for Linux and Darwin. Then these options will lead to errors for other platforms.

Reply:
> You can make these options TargetSpecific and only handle them for Linux and Darwin.

@MaskRay It looks like they are already declared as TargetSpecific - see https://github.com/llvm/llvm-project/blob/41cde46/clang/include/clang/Driver/Options.td#L4337-L4357. Please let me know if I'm missing smth and you meant smth else.

Reply:
I'm afraid that you missed the point. Keep the code in Clang.cpp but do the following

```
// Clang.cpp
if (Linux pauth abi or isDarwin) {
  claim these pauth options
}
```

Reply:
@MaskRay This would explicitly require adding each platform that would want to add pauth here. I don't think this is something desired (do we really want to add further, say, OpenBSD, FreeBSD and other platforms in the condition?)

Reply:
Yes for such straightforward CC1 forwarding options. The alternative requires a new function in CommonArgs.cpp and changes to various ToolChains/XX.cpp files, which is worse.

Reply:
@MaskRay Applied your suggestion in 346d1ef9d0d638ddda2b48a9c629f24a4385df02

> I'm afraid that you missed the point. Keep the code in Clang.cpp but do the following
> 
> ```
> // Clang.cpp
> if (Linux pauth abi or isDarwin) {
>   claim these pauth options
> }
> ```